From c680bd977aef8c1984ffff79ba78b6cc2db0945b Mon Sep 17 00:00:00 2001
From: John Lotoski <john.lotoski@iohk.io>
Date: Mon, 9 Nov 2020 17:03:57 -0500
Subject: [PATCH] Enables mantis-explorer builds from mantis-ops

- Change "yarn build" to produce a development version instead of a
  production version. Some part of the optimization process in the
  production build causes "yarn build" to fail. This is intended to be
  a short-term solution only.
- Make the "WEB3_PROVIDER" setting configurable via Nix.
---
 default.nix      | 5 +++--
 scripts/build.js | 4 ++--
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/default.nix b/default.nix
index 6614ca0..558f809 100644
--- a/default.nix
+++ b/default.nix
@@ -2,14 +2,15 @@
 #      but can (and should) be removed (along with `nix/*`) once mantis-ops/darwin
 #      is merged. kthxbye
 { src    ? ./.
-, system ? builtins.currentSystem }:
+, system ? builtins.currentSystem
+, web3Provider ? "/rpc/node" }:
 
 with import ./nix { inherit system; };
 
 mkYarnPackage {
   src       = mkSrc ./.;
 
-  WEB3_PROVIDER = "/rpc/node";
+  WEB3_PROVIDER = web3Provider;
 
   doCheck    = true;
   checkPhase = "yarn test --coverage --ci";
diff --git a/scripts/build.js b/scripts/build.js
index 76a3378..e5bb7e1 100644
--- a/scripts/build.js
+++ b/scripts/build.js
@@ -1,8 +1,8 @@
 'use strict';
 
 // Do this as the first thing so that any code reading it knows the right env.
-process.env.BABEL_ENV = process.env.BABEL_ENV || 'production';
-process.env.NODE_ENV =  process.env.NODE_ENV  || 'production';
+process.env.BABEL_ENV = process.env.BABEL_ENV || 'development';
+process.env.NODE_ENV =  process.env.NODE_ENV  || 'development';
 
 // Makes the script crash on unhandled rejections instead of silently
 // ignoring them. In the future, promise rejections that are not handled will
-- 
2.25.4

